File: problems/1275-find-winner-on-a-tic-tac-toe-game/solution.py
class Solution:
    def tictactoe(self, moves: List[List[int]]) -> str:
        grid = [['' for _ in range(3)] for _ in range(3)]
        A = 'A'
        B = 'B'
        turn = A
        for move in moves:
            row, col = move
            if turn == A:
                grid[row][col] = 'X'
                turn = B
            else:
                grid[row][col] = 'O'
                turn = A
        
        def check_winner(player):
            symbol = 'X' if player == A else 'O'
            for i in range(3):
                if all(grid[i][j] == symbol for j in range(3)):
                    return True
                if all(grid[j][i] == symbol for j in range(3)):
                    return True
            if all(grid[i][i] == symbol for i in range(3)):
                return True
            if all(grid[i][2 - i] == symbol for i in range(3)):
                return True
            return False
        
        if check_winner(A):
            return "A"
        if check_winner(B):
            return "B"
        
        if len(moves) == 9:
            return "Draw"
        else:
            return "Pending"

File: problems/1275-find-winner-on-a-tic-tac-toe-game/solution.cpp
class Solution {
public:
    string tictactoe(vector<vector<int>>& moves) {
        vector<vector<char>> board(3, vector<char>(3, ' '));
        char player = 'A';
        for (auto& move : moves) {
            board[move[0]][move[1]] = (player == 'A') ? 'X' : 'O';
            player = (player == 'A') ? 'B' : 'A';
        }

        for (int i = 0; i < 3; ++i) {
            if (board[i][0] != ' ' && board[i][0] == board[i][1] && board[i][1] == board[i][2]) {
                return (board[i][0] == 'X') ? "A" : "B";
            }
            if (board[0][i] != ' ' && board[0][i] == board[1][i] && board[1][i] == board[2][i]) {
                return (board[0][i] == 'X') ? "A" : "B";
            }
        }

        if (board[0][0] != ' ' && board[0][0] == board[1][1] && board[1][1] == board[2][2]) {
            return (board[0][0] == 'X') ? "A" : "B";
        }
        if (board[0][2] != ' ' && board[0][2] == board[1][1] && board[1][1] == board[2][0]) {
            return (board[0][2] == 'X') ? "A" : "B";
        }

        if (moves.size() == 9) {
            return "Draw";
        } else {
            return "Pending";
        }
    }
};